src/com/lilithsthrone/controller/xmlParsing/Element.java

```diff
@@ -0,0 +1,215 @@
+    /**
+     * Returns inner element. Used to support legacy code
+     * 
+     * @return w3c DOM Element that wrapper contains
+     * @deprecated 
+     */
+    @Deprecated
+    public org.w3c.dom.Element getInnerElement(){
+	return innerElement;
+    }
+    
+    /**
+     * Returns the document. Used for legacy code
+     *
+     * @return w3c DOM Document element is in.
+     * @deprecated 
+     */
+    @Deprecated
+    public org.w3c.dom.Document getDocument(){
+	return document;
+    }
+    /**
+     * <p>
+     * Returns first child element with a given tag, throwing a
+     * {@link XMLMissingTagException} if no such tag found. Used when
+     * document is invalid without element.</p>
+     * <p>
+     * Thrown exception provides the information for which tag in which file
+     * causes the error, so only one catch is needed to get all of the
+     * details.</p>
+     *
+     * @param tag String tag of the element to return.
+     *
+     * @return First element found with given tag
+     * @throws XMLMissingTagException
+     */
+    public Element getMandatoryFirstOf(String tag) throws XMLMissingTagException {
+	org.w3c.dom.Element firstElement = (org.w3c.dom.Element) innerElement.getElementsByTagName(tag).item(0);
+	if (firstElement == null) {
+	    throw new XMLMissingTagException(tag, fileDirectory);//throw error if there's no elements found
+	}
+	return new Element(firstElement, fileDirectory, document);
+    }
+
+    /**
+     * <p>
+     * Returns first child element with a given tag, or empty
+     * {@link java.util.Optional} if none are found.</p>
+     * <p>
+     * Use it when document is still valid if element is missing. Use
+     * {@link java.util.Optional#ifPresent(java.util.function.Consumer)} to
+     * create blocks that are executed only when element with a given tag is
+     * found, for example: </p>
+     * <pre>
+     *{@code
+     * parentElement.getOptionalFirstOf("extraDescription").ifPresent(
+     *	    (Element e) ->{
+     *		System.out.println(e.getTextContent);
+     *	    }
+     * );}
+     * </pre>
+     *
+     * @param tag String tag of the element to return.
+     *
+     * @return First element found with given tag.
+     */
+    public Optional<Element> getOptionalFirstOf(String tag) {
+	org.w3c.dom.Element foundElement = (org.w3c.dom.Element) innerElement.getElementsByTagName(tag).item(0);
+	if (foundElement == null) {
+	    return Optional.empty(); //return empty Optional if there's no elements found
+	}
+	return Optional.of(new Element(foundElement, fileDirectory, document));
+    }
+
+    /**
+     * <p>
+     * Returns iterable list of all child elements with a given tag</p>
+     *
+     * @param tag String tag of elements to return.
+     *
+     * @return {@link java.util.List} of Elements with given tag.
+     */
+    public List<Element> getAllOf(String tag) {
+	org.w3c.dom.NodeList nl = innerElement.getElementsByTagName(tag);
+	List<Element> returnList = new LinkedList<>();
```

Comment:
Would recommend ArrayList instead; insertion into arraylist is O(1), retrieval is O(1), iteration over the entire list is O(n). We're not going to remove elements.
You can initialize an ArrayList's initial capacity by feeding it an int, so nl.getLength() would be nice for that.

Unless you have specific reason for using LinkedList?